cli/azd/pkg/auth/manager.go

```diff
@@ -365,8 +379,10 @@ func (m *Manager) newCredentialFromClientSecret(
 	options := &azidentity.ClientSecretCredentialOptions{
 		ClientOptions: policy.ClientOptions{
 			Transport: m.httpClient,
+			Cloud:     m.cloud.Configuration,
```

Comment:
Feels like we could move the `policy.ClientOptions` as injectable that automatically handles the cloud configuration vs have to specify this in multiple places.